### PR TITLE
feature/proxy-auth

### PR DESCRIPTION
**Pull request by jrouzierinverse**
Description
-----------
New Source that allows users to log in using an http proxy

Impacts
-----------
Admin GUI login

NEWS file entries
-------

New Features
++++++++++++
* New source for logging into the Admin GUI using an http proxy

Delete branch after merge
-------------------------
NO

### CODE REVIEW

**Comment by jrouzierinverse** [comment]
Review me

**Comment by julsemaan** [comment]
This needs a documentation section in the admin guide

File: html/pfappserver/lib/pfappserver/Authentication/Credential/Proxy.pm [hunk 1]
```diff
@@ -0,0 +1,87 @@
+package pfappserver::Authentication::Credential::Proxy;
+
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Authentication/Credential/Proxy.pm:2]: Missing perldoc on subs

File: html/pfappserver/lib/pfappserver/Form/Config/Authentication/Source/AdminProxy.pm [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+use strict;
+use warnings;
+use HTML::FormHandler::Moose;
+extends 'pfappserver::Form::Config::Authentication::Source';
+
+has_block definition => (
+    render_list => [qw(proxy_addresses user_header group_header)],
+);
+
+has_field 'proxy_addresses' => (
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/Form/Config/Authentication/Source/AdminProxy.pm:26]: Can there be multiple addresses ? If so, this needs a validator as well as help text on how to separate the addresses

File: html/pfappserver/lib/pfappserver/I18N/i_default.po [hunk 3]
```diff
@@ -1 +0,0 @@
-en.po
```
Comment (julsemaan) [inline html/pfappserver/lib/pfappserver/I18N/i_default.po]: Not sure that you should modify this to put the strings back in it. My understanding was that en.po was the only place where they needed to be

File: lib/pf/Authentication/Source/AdminProxySource.pm [hunk 4]
```diff
@@ -0,0 +1,153 @@
+package pf::Authentication::Source::AdminProxySource;
```
Comment (julsemaan) [inline lib/pf/Authentication/Source/AdminProxySource.pm:1]: I might be wrong, but I don't see anything preventing this source from being put on a portal profile by mistake

If its only for use in the admin, it should be removed from the portal profiles sources choices

**Review by julsemaan — CHANGES_REQUESTED** [review]

File: html/pfappserver/lib/pfappserver/I18N/i_default.po (hunk 3 above)
Comment (jrouzierinverse) [reply html/pfappserver/lib/pfappserver/I18N/i_default.po]: That was a rebase issue I will fix that

**Comment by jrouzierinverse** [comment]
Bump

**Comment by julsemaan** [comment]
Please rebase onto devel and I'll take a look at it for the required changes

**Comment by jrouzierinverse** [comment]
Done

**Pull request merged**